pandas/core/resample.py

```diff
@@ -1696,6 +1696,9 @@ def _get_timestamp_range_edges(
     index_tz = first.tz
     if isinstance(origin, Timestamp) and (origin.tz is None) != (index_tz is None):
         raise ValueError("The origin must have the same timezone as the index.")
+    if origin == "epoch":
```

Comment:
Nit: Could we put this whole block (up to `index_tz = first.tz` even) under the `if isinstance(freq, Tick)` statement?
Seems that's where `index_tz` and `origin` only really apply.

```
    index_tz = first.tz
    if isinstance(origin, Timestamp) and (origin.tz is None) != (index_tz is None):
        raise ValueError("The origin must have the same timezone as the index.")
    if origin == "epoch":
        # set the epoch based on the timezone to have similar result between timezones
        origin = Timestamp("1970-01-01", tz=index_tz)
```

Reply:
Could we also make this an `elif`?

Reply:
Yep you are entirely right :slightly_smiling_face:. It should be fixed by now.